libtracepoint-decode-cpp/src/PerfDataFile.cpp

```diff
@@ -875,23 +874,21 @@ PerfDataFile::GetSampleEventInfo(
         }
 
         assert(iArray <= cArray);
-        pInfo->id = id;
         pInfo->event_desc = &eventDesc;
         pInfo->session_info = &m_sessionInfo;
-        pInfo->read_values = infoReadValues;
-        pInfo->callchain = infoCallchain;
+        pInfo->header = pEventHeader;
+        pInfo->id = id;
```

Comment:
Note: Just rearranging how the info struct gets filled-out.
- `id` field moved down to match the order of the fields in the struct.
- Added initialization of the new `header` field.
- The `read_values` and `callchain` field initialization is inlined into the code where they are parsed instead of being down in the "always set" section.